Unify card components: chime cards, folder cards, and question cards

jxjj opened the pull request:
This is ready for another set of eyes and some more in depth testing, but probably not ready for merge yet... I'm already seeing a few quirks that need to be fixed. So, just a draft PR for discussion for now.

--

This creates a Card component that is reused to give chime cards, folder cards, and question cards a similar look and feel.  

## New Components

- `Card` - base card component.

- `ChimeCard` - Card component specific to chimes.
<img width="1150" alt="Screen Shot 2021-11-27 at 5 23 36 PM" src="https://user-images.githubusercontent.com/980170/143723969-35b409e1-f21b-46b3-9ce8-3c768bbefa74.png">

- `FolderCard` - Card component specific to folders
<img width="1198" alt="Screen Shot 2021-11-27 at 5 23 49 PM" src="https://user-images.githubusercontent.com/980170/143723983-4b8e084a-b25b-4628-bd63-2317dc7fe4f1.png">

- `QuestionCard` - Card component specific to questions
<img width="1217" alt="Screen Shot 2021-11-27 at 5 30 01 PM" src="https://user-images.githubusercontent.com/980170/143724005-db82c1bb-0b63-427b-8e6f-5f2f202f6c9f.png">

- `CardActionButton` - Labelled button (or link) that can appear within a card, and associated with an event handler on click.
<img width="55" alt="Screen Shot 2021-11-27 at 6 07 08 PM" src="https://user-images.githubusercontent.com/980170/143724022-0fac6371-cad4-40cc-8e2d-58ab84141dc0.png">

- `Chip` - reusable "chip" (aka "badge" or "pill")
<img width="99" alt="Screen Shot 2021-11-27 at 6 07 36 PM" src="https://user-images.githubusercontent.com/980170/143724028-00631fb0-4cab-4cff-af9d-a94ccfae1c07.png">

- `DetailsItem` - for styling card details like "Join Code: 12345"
<img width="305" alt="Screen Shot 2021-11-27 at 6 09 18 PM" src="https://user-images.githubusercontent.com/980170/143724050-96f63705-a871-4dd4-8ce8-048cb93802b4.png">

- `Toggle` - Simple checkbox component that looks like a toggle. Replaces pretty-checkbox, which was more difficult to style.
<img width="60" alt="Screen Shot 2021-11-27 at 6 10 04 PM" src="https://user-images.githubusercontent.com/980170/143724065-0b1fde6d-6f05-4ed1-b4c0-b3d83e49305a.png"><img width="59" alt="Screen Shot 2021-11-27 at 6 09 59 PM" src="https://user-images.githubusercontent.com/980170/143724074-53664ed2-d63e-4974-9caa-92bd17c1865e.png">

Also:
- On the home page and chime page, a Chime will show a "Canvas" chip if associated with Canvas. The displayed join link will be the link to the Canvas site.  The Canvas chip was removed from individual folders as it could be confusing for users if a Chime shows that it's associated with Canvas, but the folder doesn't because it's not initialized yet. Resolves #61.
<img width="512" alt="Screen Shot 2021-11-27 at 6 14 20 PM" src="https://user-images.githubusercontent.com/980170/143724135-167904d6-3014-4df0-850f-aebdc8f7fa55.png">
<img width="749" alt="Screen Shot 2021-11-27 at 6 12 15 PM" src="https://user-images.githubusercontent.com/980170/143724139-7bb08c5f-96cf-44d8-b6e3-48f7ef745ac3.png">

- QuestionCard shows choices on multiple choice question and an image thumbnail on ImageHeatmap question.
<img width="1192" alt="Screen Shot 2021-11-27 at 5 35 15 PM" src="https://user-images.githubusercontent.com/980170/143724151-d49a414f-ed42-4a18-bc25-92b26628badb.png">

- Tests updated.

- FolderPage now shows a spinner while waiting for question and session info requests to resolve (prevents issue when user tries to add question before page is ready)
- moves some dev dependencies to dependencies if they're required by runtime
- "Add Question/Folder/Chime" appear consistently above the list of items throughout the app. Previously was above (Add Chime), below (Add Folder), and in settings menu (Add Question).
- Folder settings bar icons on the folder page is moved in front of text rather than behind.  Icons leading text is a more common design pattern  and associated with  [better UX for left-to-right languages](https://uxmovement.com/buttons/where-to-place-icons-next-to-button-labels/). It also matches how icons already appear on chime page (and visually looks better imho, as they read more like "bullets" in a list).
<img width="281" alt="Screen Shot 2021-11-27 at 6 18 17 PM" src="https://user-images.githubusercontent.com/980170/143724207-7c716f87-88d0-479e-bd96-efcdbb2d3492.png">
<img width="585" alt="Screen Shot 2021-11-27 at 6 18 28 PM" src="https://user-images.githubusercontent.com/980170/143724204-adab215e-5bc4-444a-8a71-3eb9caed462f.png">

Closes #89.

cmcfadden commented on resources/assets/js/views/FolderPage/QuestionCard.vue:
```diff
@@ -0,0 +1,249 @@
+import Toggle from "../../components/Toggle.vue";
+import QuestionForm from "../QuestionForm/QuestionForm.vue";
+import MultipleChoiceDisplay from "../../components/MultipleChoice/MultipleChoiceDisplay.vue";
+import HeatmapResponseDisplay from "../../components/ImageHeatmapResponse/ImageHeatmapResponseDisplay.vue";
```
Let's give these more specific names. CardDisplay or something.

cmcfadden commented on resources/assets/js/views/FolderPage/QuestionCard.vue:
```diff
@@ -0,0 +1,249 @@
+          );
+      }
+    },
+    pluralize(str, count) {
```
we have a pluralize filter globally I think?

jxjj replied:
Filters! Brilliant!!! 

Hah – I totally forgot about filters in Vue. Thanks for the suggestion.

jxjj commented:
Changes:
- Removed arrow signifier on cards
- using pluralize filter
- `DetailItem` value's like "Access Code" are highlightable on click.
- Toggle has pointer cursor.
- Updated test snapshot images

I suspect the QA team may find a few more quirks, but pretty happy where this is landing. Thanks again for the help and feedback.

cmcfadden commented:
dig it. Let's get it merged and poke around
